src/CargoBuild/CargoBuildTask.cs

```diff
@@ -0,0 +1,437 @@
+            }
+            catch (Exception ex)
+            {
+                LogException(ex);
+                return false;
+
+                void LogException(Exception ex)
+                {
+                    if (ex is AggregateException aggEx)
+                    {
+                        foreach (Exception innerEx in aggEx.InnerExceptions)
+                        {
+                            LogException(innerEx);
+                        }
+                    }
+                    else
+                    {
+                        Log.LogErrorFromException(ex, showStackTrace: true);
+                    }
+                }
+            }
+            finally
+            {
+                if (_shouldCleanRustPath)
+                {
+                    CleanupRustPath();
+                }
+            }
+        }
+
+        private async Task<ExitCode> RustFetchAsync(string workingDir, bool authorize = false)
+        {
+            ExitCode authResult = authorize ? await DoRegistryAuthAsync(workingDir) : ExitCode.Succeeded;
+
+            if (authorize && authResult == ExitCode.Succeeded || !authorize)
+            {
+                string path;
+                string args;
+                const int RetryCnt = 2;
+
+                path = _cargoPath;
+                args = "fetch";
+
+                Log.LogMessage(MessageImportance.Normal, $"Fetching cargo crates for project in {workingDir}");
+                var exitCode = await ExecuteWithRetriesAsync(path, processArgs: args, workingDir, retryCount: RetryCnt, processRetryArgs: args);
+                Log.LogMessage(MessageImportance.Normal, $"Finished fetching cargo crates for project in  {workingDir}");
+                return exitCode;
+            }
+
+            return authResult;
+        }
+
+        private async Task<ExitCode> ExecuteWithRetriesAsync(string processFileName, string processArgs, string workingDir, int retryCount, string? processRetryArgs = null)
+        {
+            ExitCode exitCode = await ExecuteProcessAsync(processFileName, processArgs, workingDir);
+            const int InitialWaitTimeSec = 3;
+            int retries = 0;
+            while (exitCode != 0 && retries < retryCount)
+            {
+                retries++;
+                int wait = InitialWaitTimeSec * 1000 * retries;
+                Log.LogMessage(MessageImportance.Normal, $"Process failed with exit code: {exitCode}. Retry #{retries}: {processFileName}. Waiting {wait / 600} seconds before retrying.");
+
+                await Tasks.Task.Delay(wait);
+                exitCode = await ExecuteProcessAsync(processFileName, processRetryArgs ?? processArgs, workingDir);
+            }
+
+            return exitCode;
+        }
+
+        private async Task<ExitCode> DoRegistryAuthAsync(string workingDir)
+        {
+            return await ExecuteWithRetriesAsync(_cargoPath, $"login --registry {RegistryFeedName}", workingDir, retryCount: 2);
```

Comment:
How does auth work with cargo? Will this pop up an auth window? Does it require user input on the console?